### DISCUSSION TITLE
How to add headers to oicd oauth2 flow ? (Oauth error 900902)

### DISCUSSION THREAD

**Discussion by STaX-62** [pocketbase/pocketbase#4406] [Q&A]
Hello,

i'm having an issue while trying to use OpenId connect, i'm getting this :

ERROR POST /api/collections/users/auth-with-oauth2
└─ Something went wrong while processing your request.
   └─ oauth2: "900902"

I don't really know where I messed up, I've been searching for a while and i found that it could could be a credential issue but I have no idea on how I could resolve that issue.

Would appreciate if someone could help me on this one!

**Comment by ganigeorgiev** [comment]
I don't think anyone will be able to help you without details what OIDC vendor you use and how you've configured it.

Googling the error "900902" code leads to [WSO2](https://apim.docs.wso2.com) and from [their documentation](https://apim.docs.wso2.com/en/latest/troubleshooting/error-handling/) the error code refers to:

> Missing credentials
> No authentication information provided | Accessing an API without Authorization: Bearer header

If that's the vendor you use, double check its configuration, permissions/scopes and whether you use the correct auth, token and userinfo urls.

**Reply by STaX-62** [reply]
Okay, I think I found where the issue is, i need to add multiple headers in all requests I send to the vendor to make it work.
I tried to add them with js sdk for now: 
pb.collection('users').authWithOAuth2({
    provider: 'oidc', 'headers': {
         //here
    }
  })
 but I doesn't seems to work and I think it should be on the server side but i don't know I can add that.

how could i pass these headers to my auth, token et userinfo requests ?
